pkg/controllers/managementuser/rbac/prtb_handler_test.go

```diff
@@ -186,3 +192,273 @@ func Test_manager_checkForGlobalResourceRules(t *testing.T) {
+								Resources:     []string{"myresource"},
+								Verbs:         []string{"delete", "list"},
+							},
+						},
+					}, in1)
+					return nil, nil
+				},
+			},
+			want: "myrole-promoted",
+		},
+		{
+			name: "removing verbs from policy will remove the rule",
+			args: args{
+				rtName:   "myrole",
+				resource: "myresource",
+				newVerbs: sets.New[string](),
+				baseRule: rbacv1.PolicyRule{
+					APIGroups:     []string{"management.cattle.io"},
+					ResourceNames: []string{"local"},
+				},
+			},
+			crListerMock: &typesrbacv1fakes.ClusterRoleListerMock{
+				GetFunc: func(namespace, name string) (*v1.ClusterRole, error) {
+					return &rbacv1.ClusterRole{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "myrole-promoted",
+						},
+						Rules: []rbacv1.PolicyRule{
+							{
+								APIGroups:     []string{"another.cattle.io"},
+								ResourceNames: []string{"foobar"},
+								Resources:     []string{"nodes"},
+								Verbs:         []string{"create"},
+							},
+							{
+								APIGroups:     []string{"management.cattle.io"},
+								ResourceNames: []string{"local"},
+								Resources:     []string{"myresource"},
+								Verbs:         []string{"get", "list"},
+							},
+						},
+					}, nil
+				},
+			},
+			clusterRolesMock: &typesrbacv1fakes.ClusterRoleInterfaceMock{
+				CreateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Fail(t, "unexpected call to ClusterRole Create")
+					return in1, nil
+				},
+				UpdateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Equal(t, &rbacv1.ClusterRole{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "myrole-promoted",
+						},
+						Rules: []rbacv1.PolicyRule{
+							{
+								APIGroups:     []string{"another.cattle.io"},
+								ResourceNames: []string{"foobar"},
+								Resources:     []string{"nodes"},
+								Verbs:         []string{"create"},
+							},
+						},
+					}, in1)
+					return in1, nil
+				},
+			},
+			want: "myrole-promoted",
+		},
+		{
+			name: "reconciling non existing ClusterRole with no verbs is nop-op",
+			args: args{
+				rtName:   "myrole",
+				resource: "myresource",
+				newVerbs: sets.New[string](),
+				baseRule: rbacv1.PolicyRule{
+					APIGroups:     []string{"management.cattle.io"},
+					ResourceNames: []string{"local"},
+				},
+			},
+			crListerMock: &typesrbacv1fakes.ClusterRoleListerMock{
+				GetFunc: func(namespace, name string) (*v1.ClusterRole, error) {
+					return nil, errNotFound
+				},
+			},
+			clusterRolesMock: &typesrbacv1fakes.ClusterRoleInterfaceMock{
+				CreateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Fail(t, "unexpected call to ClusterRole Create")
+					return in1, nil
+				},
+				UpdateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Fail(t, "unexpected call to ClusterRole Update")
+					return in1, nil
+				},
+			},
+			want: "",
+		},
+		{
+			name: "create a the role if get fails",
```

Comment:
```suggestion
			name: "create the role if get fails",
```
As above, I think the generic failure and "not found" should be separated https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#NewNotFound might help :-)

Reply:
Fix 678f8a1